client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,72 @@
+import { ChangeEvent } from 'react';
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Controller that adds the ability to toggle some or all the elements, ignoring the disabled ones
+ * Additionaly, it checks if all non-disabled items are selected or not. If they are, then select all
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-w-bulk-target="all">
+ *
+ *   <div>
+ *     <input type="checkbox" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *   </div>
+ *
+ *   <button data-action="w-bulk#clear">Clear</button>
+ * </div>
+ */
+export class BulkController extends Controller {
+  /** The list of targets for this controller. */
+  static targets = ['all', 'item'];
+
+  /** The list of all select-all checkbox targets for this controller. */
+  declare readonly allTargets: HTMLInputElement[];
+
+  /** The list of all item checkbox targets for this controller. */
+  declare readonly itemTargets: HTMLInputElement[];
+
+  /**
+   * Toggles the select-all checkbox based on item checkboxes.
+   */
+  toggle(): void {
+    // Check if all non-disabled item checkboxes are checked
+    const nonDisabledItemTargets = this.itemTargets.filter(
+      (item) => !item.disabled,
+    );
+    const isAllChecked = !nonDisabledItemTargets.some((item) => !item.checked);
+
+    // Update all select all checkboxes (except disabled ones)
+    this.allTargets.forEach((target) => {
+      if (!target.disabled) {
+        // eslint-disable-next-line no-param-reassign
+        target.checked = isAllChecked;
+      }
+    });
+  }
+
+  /**
+   * Toggles all item checkboxes based on select-all checkbox.
+   */
+  toggleAll(event: ChangeEvent<HTMLInputElement>): void {
+    const isChecked = event.target.checked;
+
```

Comment:
```suggestion
  toggleAll(event: Event | {params?: { force: boolean }} ): void {
    const isChecked = typeof force === 'boolean' ? force: event.target.checked;

```

Code may not be correct but general gist is that we allow for the `params` to come in via Stimulus (see note above with the button usage). If there is a boolean value for `params.force` then use it, otherwise read the `target.checked` value.

see https://stimulus.hotwired.dev/reference/actions#action-parameters